sig/types.rbs

```diff
@@ -301,16 +301,21 @@ module RBS
     end
 
     class Record
-      attr_reader fields: Hash[Symbol, t]
+      attr_reader all_fields: Hash[Symbol, [t, bool]]
 
       type loc = Location[bot, bot]
 
       def initialize: (fields: Hash[Symbol, t], location: loc?) -> void
+                    | (all_fields: Hash[Symbol, [t, bool]], location: loc?) -> void
 
       include _TypeBase
 
       attr_reader location: loc?
 
+      def fields: () -> Hash[Symbol, t]
+
+      def optional_fields: () -> Hash[Symbol, t]?
```

Comment:
Ah, I made a mistake to put `?` at the end of the method type. It should return an empty hash.

```suggestion
      def optional_fields: () -> Hash[Symbol, t]
```

Reply:
if you look at the implementation, nothing is returned if all_fields is the same as fields. I think that's fine, i.e. we can avoid creating that extra hash when not necessary. Or do you see any use for it?